gateleen-core/src/main/java/org/swisspush/gateleen/core/util/HttpHeaderUtil.java

```diff
@@ -61,18 +68,43 @@
      * Helper method to check the presence of a http header matching the provided pattern. Every header entry is checked
      * in the format KEY: VALUE
      *
-     * @param headers   The Map with the header key - value pairs to be evaluated
+     * @param headers        The Map with the header key - value pairs to be evaluated
      * @param headersPattern The pattern to match the headers against.
      * @return True if a matching header was found, false otherwise
      */
     public static <T extends MultiMap> boolean hasMatchingHeader(@Nonnull T headers, @Nonnull Pattern headersPattern) {
         for (Map.Entry<String, String> entry : headers) {
             String header = entry.getKey() + ": " + entry.getValue();
-            if(headersPattern.matcher(header).matches()){
+            if (headersPattern.matcher(header).matches()) {
                 return true;
             }
         }
         return false;
     }
 
+    /**
+     * Merges headers, makes sure that only one value of header {@link HttpRequestHeader#CONTENT_LENGTH} ends up in the result.
+     */
+    public static void mergeHeaders(@Nonnull MultiMap destination, @Nonnull MultiMap source, @Nullable String context) {
```

Comment:
I think would be helpful to document at least the parameter "*context*". I think something like the following could clarify things for callers:
> @param context  Only used for log/error messages. Does not impact how we merge.

Reply:
Right, will improve, thx.